app/components/data/action/add/file-group-create-form.component.ts

```diff
@@ -0,0 +1,152 @@
+            name: ["", [
+                Validators.required,
+                Validators.maxLength(validation.maxLength.fileGroup),
+                Validators.pattern(validation.regex.fileGroup),
+            ], [
+                this._validateFileGroupName.bind(this),
+            ]],
+            folder: ["", [Validators.required]],
+            includeSubDirectories: [true],
+            options: [null, []],
+            accessPolicy: ["private"],
+        });
+    }
+
+    public ngOnInit() {
+        this.editing = false;
+    }
+
+    @autobind()
+    public submit(): Observable<any> {
+        const formData = this.getCurrentValue();
+        const batch = Constants.ResourceUrl.batch;
+        const arm = Constants.ResourceUrl.arm;
+
+        // We now need to pass in both the Batch and ARM tokens as we are dealing with storage.
+        const observable = this.accountService.currentAccount.cascade((account) => {
+            return this.adalService.accessTokenFor(account.subscription.tenantId, batch).cascade((batchToken) => {
+                return this.adalService.accessTokenFor(account.subscription.tenantId, arm).cascade((armToken) => {
+                    return this.pythonRpcService.call("create_file_group", [
+                        batchToken,
+                        armToken,
+                        formData.name,
+                        formData.includeSubDirectories ? path.join(formData.folder, "**\\*") : formData.folder,
+                        formData.options,
+                        account.toJS(),
+                    ]).catch((error) => {
+                        return Observable.throw(ServerError.fromPython(error));
+                    });
+                });
+            });
+        }).share();
+
+        observable.subscribe({
+            next: (data) => {
+                const message = `${data.uploaded} files were successfully uploaded to the file group`;
+                this.storageService.onFileGroupAdded.next(`${this.storageService.ncjFileGroupPrefix}${formData.name}`);
+                this.notificationService.success("Create file group", message, { persist: true });
+            },
+            error: (error) => {
+                log.error("Failed to create form group", error);
+            },
+        });
+
+        return observable;
+    }
+
+    public dtoToForm(fileGroup: FileGroupCreateDto): CreateFileGroupModel {
+        this.editing = true;
+        this.title = "Add folder to file group";
+        this.description = "Add another folder to an already existing file group";
+        this.form.controls.name.disable();
+
+        return fileGroupToFormModel(fileGroup);
+    }
+
+    public formToDto(data: CreateFileGroupModel): FileGroupCreateDto {
+        return createFileGroupFormToJsonData(data);
+    }
+
+    @autobind()
+    public selectFolder(changeEvent: Event) {
+        const element = changeEvent.srcElement as any;
+        this.form.controls["folder"].markAsTouched();
```

Comment:
save the folder control separatly when creating the form.

this.folderControl = formBuilder.control(...);

this.form = formBuilder.group({
...
folder: this.folderControl,
...
});